rust/kernel/time.rs

```diff
@@ -0,0 +1,15 @@
+// SPDX-License-Identifier: GPL-2.0
+
+//! This module contains timing methods for functionality like sleeping.
```

Comment:
Similar to above, this could be written like:

```suggestion
//! Timing methods for functionality like sleeping.
```

However, unless there is a good reason, it is best to follow the C side and call this the `delay` module instead, which seems a better name for these functions anyway, and makes it consistent with the C side.

Also, please add a reference like we do in other modules, e.g.

```rust
//! C header: [`include/linux/delay.h`](../../../../include/linux/delay.h)
```

Reply:
What do you mean by the C side? Do you mean the headers as opposed to the file where the method is defined? I saw it was defined in `time.c` but declared in `delay.h`. If the headers are the better option, I can keep that in mind.

Reply:
I think you mean `timer.c`, but it is also true it is in a `time` folder :)

Sometimes the C side implements things in a different way that it is structured in the headers, indeed, but in general I think following the headers is better, since those are what consumers/users are supposed to be using, especially if they are better organized.

In this case, the `timer.c` seems to implement bits for both `timer.h` and `delay.h`. Since it is already separated in the headers (and the separation makes sense), I would follow that.

Of course, this is not to say we should always do that! In fact, sometimes we may want to do things differently than the C side. But here, since at least at the moment it is just for a delay function, and `delay.h` exists, I would call it that.

Reply:
Okay that makes sense! And yes, I was looking at the directory it looks like. :)